### ISSUE TITLE
Modify fan speed via HASS-miner

### ISSUE THREAD

**Issue by UpstreamData** [Schnitzel/hass-miner#170]
There have been some requests to be able to control fixed fan speed via HASS-miner.  After the merge of #39, this requires:

- A switch entity to enable or disable manual fan control,
- and a number entity to set manual fan speed.

This would then have to call `miner.get_config()` on update, then depending on switch status:

- Set `cfg.temp_mode` to `"manual"`/`"disabled"` or `"auto"` depending on the switch,
- and set `cfg.fan_speed` to an integer between 1 and 100.

**Comment by Frenkel85** [comment]
This would be brilliant! Was looking for exactly this. Would be great if it will be integrated

**Comment by Domek688** [comment]
Yes, it would be excellent if we could control the fan speed from the integration. Or at least throttle them down to a low run when the BOS is stopped. This would significantly reduce the power consumption of the machine. As for power management, it would also be excellent if it were possible to control (on/off) individual hash boards. This would manage the power consumption perhaps better than if there is only the possibility to set 2450 to 3200VA range (with a standard power supply). 
Anyway, thank you Schnizel for a great job. I used to control the Miner S19 PRO remotely only by switching the contactor. I have destroyed 3pcs APW power supply in two years, because the switching power supply probably does not tolerate frequent shocks when switched on hard.

**Comment by UpstreamData** [comment]
This may complicate things somewhat.  I'm not sure setting fan speed on shutdown is possible, I know it can be set to the minimum possible speed by disabling bosminer_monitor, but that should be done 3-5 minutes after shutdown to prevent frying any boards.  Not sure how any other firmwares would handle this either; both Whatsminer and Antminer stock firmware can be shut down, and whatsminers can be tuned, but not sure how they will react with manually setting 0 speed when in shutdown mode.

Regarding disabling boards, that might be something we could do automatically?  Assume each board draws a minimum of 800W (2400/3), then shut down boards at 2400W and 1600W, although I don't know how BOS will react to this, another thing I've not tested myself.  It's possible it will try to use 2400W on that single board, but it's also possible it will split it up evenly across all boards, and ignore the disabled one.

**Comment by Domek688** [comment]
> This may complicate things somewhat. I'm not sure setting fan speed on shutdown is possible, I know it can be set to the minimum possible speed by disabling bosminer_monitor, but that should be done 3-5 minutes after shutdown to prevent frying any boards. Not sure how any other firmwares would handle this either; both Whatsminer and Antminer stock firmware can be shut down, and whatsminers can be tuned, but not sure how they will react with manually setting 0 speed when in shutdown mode.
> 
> Regarding disabling boards, that might be something we could do automatically? Assume each board draws a minimum of 800W (2400/3), then shut down boards at 2400W and 1600W, although I don't know how BOS will react to this, another thing I've not tested myself. It's possible it will try to use 2400W on that single board, but it's also possible it will split it up evenly across all boards, and ignore the disabled one.

![Dashboard S19 PRO - two board](https://github.com/Schnitzel/hass-miner/assets/148078849/c1afb08c-eb60-41d8-9146-d81fe471f43a)
![Config performace S19 PRO](https://github.com/Schnitzel/hass-miner/assets/148078849/6f38d994-a746-4f50-bc27-24993be19db7)
Hello,
- I have tested the BOS firmware with individual boards shutting down. Firmware responds correctly, performance drops by a third when one hashboard is disabled, by two thirds when two hashboards are disabled - compared to the set value, see attachment. The only anomaly is that after turning off or on a hashboard, all boards shutdown for about two minutes. After this time, however, the hashboards that are on will boot to full power.
- On the point of fan settings - I don't know about other firmware, but I'm running the S19 with BOS where the fans are still set to 100% speed - manually, as recommended by Braiins experts. This is because the chips increase power consumption with increasing temperature. For this reason I don't use automatic fan speed control. But if the BOS is not running, it would be great to at least lower the fans, say at least to 30%, this would eliminate possible overheating of the chips by accumulated heat after the BOS is turned off.
